src/backend/distributed/executor/multi_utility.c

```diff
@@ -461,6 +462,81 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 							params, dest, completionTag);
 #endif
 
+#if (PG_VERSION_NUM >= 100000)
+	if (IsA(parsetree, CreateStmt))
+	{
+		CreateStmt *createStatement = (CreateStmt *) parsetree;
+
+		/*
+		 * If a partition is being created and if its parent is a distributed
+		 * table, we will distribute this table as well.
+		 */
+		if (createStatement->inhRelations != NIL && createStatement->partbound != NULL)
+		{
+			RangeVar *parentRelation = linitial(createStatement->inhRelations);
+			bool parentMissingOk = false;
+			Oid parentRelationId = RangeVarGetRelid(parentRelation, NoLock,
+													parentMissingOk);
+
+			if (IsDistributedTable(parentRelationId))
+			{
+				bool missingOk = false;
+				Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock,
+												  missingOk);
+				Var *parentDistributionColumn = DistPartitionKey(parentRelationId);
+				char parentDistributionMethod = DISTRIBUTE_BY_HASH;
+				uint32 parentColocationId = TableColocationId(parentRelationId);
+
+				CreateDistributedTable(relationId, parentDistributionColumn,
+									   parentDistributionMethod, parentColocationId);
+			}
+		}
+	}
+	else if (IsA(parsetree, AlterTableStmt))
+	{
+		AlterTableStmt *alterTableStatement = (AlterTableStmt *) parsetree;
+		List *commandList = alterTableStatement->cmds;
+		ListCell *commandCell = NULL;
+
+		foreach(commandCell, commandList)
+		{
+			AlterTableCmd *alterTableCommand = (AlterTableCmd *) lfirst(commandCell);
+
+			/*
+			 * If a partition is being attached to a distributed table, we will distribute
+			 * attached table as well.
+			 */
+			if (alterTableCommand->subtype == AT_AttachPartition)
+			{
+				Oid relationId = AlterTableLookupRelation(alterTableStatement, NoLock);
+				PartitionCmd *partitionCommand = (PartitionCmd *) alterTableCommand->def;
+				bool partitionMissingOk = false;
+				Oid partitionRelationId = RangeVarGetRelid(partitionCommand->name, NoLock,
+														   partitionMissingOk);
+
+				if (!IsDistributedTable(relationId) &&
```

Comment:
We should never be able to hit this check since we prevent it on the table creation. I think keeping this check is OK, but maybe add a single line comment saying that we don't expect to this here but keeping it out of paranoia or so ?